airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/db/factory/DataSourceFactory.java

```diff
@@ -178,14 +178,14 @@ private static class DataSourceBuilder {
     private String driverClassName;
     private String host;
     private String jdbcUrl;
-    private int maximumPoolSize = 10;
+    private int maximumPoolSize = 5;
```

Comment:
this might affect throughput? can we do a performance harness run and measure?

Reply:
I'm not sure this helped. One of my theories was the database has a per user connection limit and we were creating too many connections and timing out a portion of the connections, which would explain why we only see this on close.